http_server.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import socket
+import email.utils
+
+
+def server_socket_function():
+    server_socket = socket.socket(socket.AF_INET,
+                                  socket.SOCK_STREAM,
+                                  socket.IPPROTO_IP)
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(1)
+
+    try:
+        while True:
+            conn, addr = server_socket.accept()
+
+            recieve_total = ""
+            buffersize = 32
+            finished = 0
+            while not finished:
+                recieve = conn.recv(buffersize)
+                if len(recieve) < buffersize:
+                    recieve_total += recieve
+                    finished = 1
+                else:
+                    recieve_total += recieve
+
+            if recieve_total:
+                try:
+                    response = parse_request(recieve_total)
```

Comment:
The `parse_request` method is supposed to return a URI.  the specification says so.  This clearly is expecting something else.